t/lib/proxy-wasm-tests/hostcalls/src/lib.rs

```diff
@@ -84,15 +112,34 @@ impl RootContext for TestRoot {
             self.get_config("tick_period").unwrap()
         );
 
+        let n_sync_calls = self
+            .config
+            .get("n_sync_calls")
+            .map_or(1, |v| v.parse().expect("bad n_sync_calls value"));
+
+        if self.n_sync_calls >= n_sync_calls {
+            return;
+        }
+
         match self.get_config("on_tick").unwrap_or("") {
             "log_property" => test_log_property(self),
+            "define_metrics" => {
+                test_define_metrics(self, TestPhase::Tick);
+                self.n_sync_calls += 1;
+            }
+            "increment_counters" => test_increment_counters(self, TestPhase::Tick, None),
+            "toggle_gauges" => test_toggle_gauges(self, TestPhase::Tick, None),
+            "set_gauges" => {
+                test_record_metric(self, TestPhase::Tick);
+                self.n_sync_calls += 1;
+            }
+            "record_histograms" => {
+                test_record_metric(self, TestPhase::Tick);
+                self.n_sync_calls += 1;
+            }
+            "log_metrics" => test_log_metrics(self, TestPhase::Tick),
```

Comment:
Same comment for moving these to the context checks crate.

Reply:
0313c03a69deea43975a08161dbb7ab7b657b9b5